pandas/core/resample.py

```diff
@@ -1139,7 +1142,8 @@ def _get_time_bins(self, ax):
                                         start=first,
```

Comment:
what exactly are you trying to do here? what is the purpose of ``values_present``, IOW what case are you trying to handle?

Reply:
the case is described here https://github.com/pandas-dev/pandas/issues/15549
unfortunately, I get `values fall after last bin` error when I do resampling of tz-aware series with summer-winter time change

Reply:
so in case if we are resampling series I believe an extra bin should be generated to handle this
but in case of generation of the datetimeindex with date_range for example we should not get the extra bin

Reply:
purpose of values_present is to be able to say which case is that, that we are creating datetimeindex for

Reply:
hope that makes sense

Reply:
I understand the case. My point is that you can detect what you need and simply generate different bins at that point, rather than intrude to the implementation of ``date_range``.

Reply:
the soln that you posted in the issue looks promising. did you try that?

Reply:
yep, last time I made a PR with it you said that it is not right to resample by UTC a tz aware series
and after giving it a thought I think you are right
it's not a good idea to resample everything by UTC also I did that change internally for our company and I got issues with it

Reply:
yeah that might be a bigger change, ok